var/spack/repos/builtin/packages/intel-oneapi-mpi/package.py

```diff
@@ -16,6 +16,8 @@ class IntelOneapiMpi(IntelOneApiLibraryPackage):
 
     homepage = 'https://software.intel.com/content/www/us/en/develop/tools/oneapi/components/mpi-library.html'
 
+    executables = [r'^mpiicpc$']  # one should be sufficient
```

Comment:
How does this differentiate between the old, non-oneapi intel mpi compiler?  Or does that not matter in this case.

Reply:
It doesn't. Component versions have diverged in the latest oneapi release. I will redo this by parsing the path of the mpiicpc, which will be a reliable way to detect mpi version number.

Reply:
The challenge with extracting the version from the path from my testing, IIRC, is it can yield a version on some systems that is not in the Spack package.  Perhaps that's a non-issue.